### ISSUE TITLE
Investigate magenta flashing screen on macOS on file load

### ISSUE THREAD

**Issue by foxnne** [foxnne/pixi#9]
- "You can create an empty texture and then fill it with `writeTexture()`. You can also create a mapped buffer, fill it on the CPU side and then copy it to the texture with `copyBufferToTexture()`"

**Comment by foxnne** [comment]
I recently added https://github.com/foxnne/pixi/blob/dd7180f030f3ca95ce9639a022774b75206a3807/src/editor/editor.zig#L72-L116
but now I'm seeing that I get random freezes with no crash and flashing magenta screen on arm macOS, and both on arm macOS and Windows the files seem to load and draw randomly. Sometimes certain layers don't draw, sometimes no layers draw at all.

**Comment by foxnne** [comment]
Okay, I've discovered that only the first layers in a file were not drawing. I've fixed this issue in a recent commit so texture loading and drawing seems to be working now. 

However, still, occasionally on macOS I get a magenta screen when opening a file after a hard freeze. I believe the magenta screen is being thrown up by dear imgui as a shader issue, but I don't know yet if it's a bug with my code or if maybe the timing of loading the file causes an issue with it.
